StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedFormViewController.swift

```diff
@@ -0,0 +1,430 @@
+    private let stackView: UIStackView = UIStackView()
+
+    weak var delegate: EmbeddedFormViewControllerDelegate?
+
+    // MARK: - Initializers
+
+    init(configuration: EmbeddedPaymentElement.Configuration,
+         loadResult: PaymentSheetLoader.LoadResult,
+         paymentMethodType: PaymentSheet.PaymentMethodType,
+         previousPaymentOption: PaymentOption? = nil,
+         analyticsHelper: PaymentSheetAnalyticsHelper,
+         formCache: PaymentMethodFormCache) {
+        self.loadResult = loadResult
+        self.intent = loadResult.intent
+        self.elementsSession = loadResult.elementsSession
+        self.configuration = configuration
+        self.previousPaymentOption = previousPaymentOption
+        self.analyticsHelper = analyticsHelper
+        self.paymentMethodType = paymentMethodType
+        self.formCache = formCache
+        
+        super.init(nibName: nil, bundle: nil)
+
+        add(childViewController: paymentMethodFormViewController, containerView: paymentContainerView)
+        updateUI()
+    }
+
+    /// Updates all UI elements (pay button, error, mandate)
+    private func updateUI() {
+        updatePrimaryButton()
+        updateMandate()
+        updateError()
+    }
+
+    private func updatePrimaryButton() {
+        let callToAction: ConfirmButton.CallToActionType = {
+            if let override = paymentMethodFormViewController.overridePrimaryButtonState {
+                return override.ctaType
+            }
+            if let customCtaLabel = configuration.primaryButtonLabel {
+                return shouldDeferConfirmation ? .custom(title: customCtaLabel) : .customWithLock(title: customCtaLabel)
+            }
+
+            if shouldDeferConfirmation {
+                return .continue
+            }
+            return .makeDefaultTypeForPaymentSheet(intent: intent)
+        }()
+        let state: ConfirmButton.Status = {
+            if isPaymentInFlight {
+                return .processing
+            }
+            if let override = paymentMethodFormViewController.overridePrimaryButtonState {
+                return override.enabled ? .enabled : .disabled
+            }
+            return selectedPaymentOption == nil ? .disabled : .enabled
+        }()
+        primaryButton.update(
+            state: state,
+            style: .stripe,
+            callToAction: callToAction,
+            animated: true
+        )
+    }
+
+    func updateMandate() {
+        let mandateProvider = VerticalListMandateProvider(configuration: configuration, elementsSession: elementsSession, intent: intent)
+        let newMandateText = mandateProvider.mandate(for: selectedPaymentOption?.paymentMethodType,
+                                                     savedPaymentMethod: selectedPaymentOption?.savedPaymentMethod,
+                                                     bottomNoticeAttributedString: paymentMethodFormViewController.bottomNoticeAttributedString)
```

Comment:
nit I've realized that Xcode's indentation style is different from this, so let's try to use that.  You can let it auto-format by highlighting code and hitting ctrl-m

Reply:
Hrm, I tried ctrl-m but it formats the file in a really odd way.

```swift
@_spi(
    STP
) import StripeCore
```

And things like

```swift
    func updateErrorLabel(
        for error: Swift.Error?
    ) {
        self.error = error
        updateError()
    }
```

IMO the linter should catch this.

Reply:
Oh yeah I wouldn't run ctrl-m on the whole file.  Just individual code blocks. Agree linter should enforce this